### ISSUE TITLE
KokkosSparse::spmv build error (-Werror) with GCC 7.2.0

### ISSUE THREAD

**Issue by mhoemmen** [kokkos/kokkos-kernels#448]
```
.../packages/kokkos-kernels/src/sparse/KokkosSparse_spmv.hpp:231:32: warning: comparison between ‘enum Kokkos::ViewTraits<const float**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >::<anonymous>’ and ‘enum Kokkos::ViewTraits<float**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >::<anonymous>’ [-Wenum-compare]
   static_assert (XVector::rank == YVector::rank,
                                ^
```
Looks easy to fix; just cast both sides to `int`.

**Comment by mhoemmen** [comment]
This is a build error in Trilinos' PR testing now:
```
/scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_7.2.0/Trilinos/packages/kokkos-kernels/src/sparse/KokkosSparse_spmv.hpp:231:32: error: comparison between ‘enum Kokkos::ViewTraits<const Kokkos::complex<double>**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<1> >::<unnamed>’ and ‘enum Kokkos::ViewTraits<Kokkos::complex<double>**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, Kokkos::MemoryTraits<1> >::<unnamed>’ [-Werror=enum-compare]
   static_assert (XVector::rank == YVector::rank,
                  ~~~~~~~~~~~~~~^~~~~~~~~~
cc1plus: all warnings being treated as errors
```
It didn't show up in Trilinos before because nobody was calling `KokkosSparse::spmv` before with an input (multi)vector that's a View of const T.  I need to patch Trilinos to fix this.

**Comment by ndellingwood** [comment]
Addressed by PR #450 by @brian-kelley, marking as InDevelop
